src/com/nisovin/magicspells/castmodifiers/conditions/UnderBlockCondition.java

```diff
@@ -0,0 +1,99 @@
+package com.nisovin.magicspells.castmodifiers.conditions;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockFace;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.entity.Player;
+
+import com.nisovin.magicspells.materials.MagicMaterial;
+import com.nisovin.magicspells.util.RegexUtil;
+import com.nisovin.magicspells.DebugHandler;
+import com.nisovin.magicspells.MagicSpells;
+import com.nisovin.magicspells.castmodifiers.Condition;
+
+public class UnderBlockCondition extends Condition {
+
+	int height;
+	String blocks;
+	
+	//Block Data
+	Set<Material> types;
+	List<MagicMaterial> mats
+
+	@Override
+	public boolean setVar(String var) {
+		//Lets TRY and catch some formatting mistakes for this modifier.
+		try {
+			String[] variable = var.split(";",2);
+			blocks = variable[0];
+			height = Integer.parseInt(variable[1]);
+		} catch (NumberFormatException e) { //Oh no, that variable[1] is somehow not a string? give them an Error!
+			DebugHandler.debugNumberFormat(e);
+			return false;
+		} catch (ArrayIndexOutOfBoundsException missingSemiColon) { //No ; in modifier? Just great, give them an Error!
+			MagicSpells.error("No ; seperator for height was found!");
+			return false;
+		}
+
+		//Checks if they put any blocks to compare with in the first place.
+		if (blocks.equals("")) {
+			MagicSpells.error("Didn't specify any blocks to compare with.");
+			return false;
+		}
+		
+		//We need to parse a list of the blocks required and check if they are valid.
+		if (blocks.contains(",")) {
```

Comment:
You don't actually need this if statement, because if you try splitting on the comma and there is no comma, it just returns the original string in an array, allowing you to still loop through it just as if there was a comma separated list.